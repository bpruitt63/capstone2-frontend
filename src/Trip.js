import React, {useState} from 'react';
import {Button} from 'reactstrap';
import Point from './Point';
import TripRatingForm from './TripRatingForm';
import { calculateDistance } from './static/helpers';
import DeleteModal from './DeleteModal';

/** Component for a single saved trip from user's trips or recent trips list */
function Trip({trip, units, username, deleteTrip, isMobile}) {

    const [showLocations, setShowLocations] = useState(false);
    const [rating, setRating] = useState();
    const distance = calculateDistance(units, trip.distance);
    const isSaved = true;

    /** Toggle whether locations for trip are visible */
    const toggleLocations = () => {
        setShowLocations(!showLocations);
    };

    return (
        <div className='Trip'>
            <div className='tripTitle'>
                <h5>
                    <span>{trip.tripName}</span>{' '}-{' '}{distance}{' '}
                    {units === 'i' ? 'mi' : 'km'}
                </h5>
                {<p>Rating: {rating || trip.tripRating || 'Not rated yet'}</p>}
            </div>
            {username &&
                <TripRatingForm trip={trip}
                                username={username}
                                setRating={setRating}
                                isMobile={isMobile} />
            }
            {showLocations &&
                trip.locations.map(l =>
                    <Point key={l.locationId}
                            point={l}
                            currentTrip={[]}
                            setCurrentTrip={{}}
                            username={username}
                            isSaved={isSaved}
                            isMobile={isMobile} />
                )}
            <Button onClick={toggleLocations} className='toggle'>
                {showLocations ? 'Hide Trip' : 'Show Trip'}
            </Button>
            <br/>
            {username &&
                <DeleteModal deleteTrip={deleteTrip}
                                username={username}
                                id={trip.tripId} />
            }
        </div>
    );
};

export default Trip;