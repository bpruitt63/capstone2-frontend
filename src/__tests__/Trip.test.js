import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Trip from '../Trip';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({}),
}));  

const trip = {tripId: 1, 
                tripName: 'test1', 
                distance: '1000', 
                tripRating: 1, 
                locations: [{locationId: "testLoc1", 
                            locationPosition: 0, 
                            locationName: 'testLocName1'},
                            {locationId: "testLoc2", 
                            locationPosition: 1, 
                            locationName: 'testLocName2'}]};

test('renders', () => {
    render(<Trip trip={trip}/>);
});

test('matches snapshot', () => {
    const {asFragment} = render(<Trip trip={trip}/>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders recent trip', () => {
    render(<Trip trip={trip}/>);
    expect(screen.getByText(/Show Trip/)).toBeInTheDocument;
    expect(screen.queryByText(/Delete Trip/)).not.toBeInTheDocument;
});

test('renders user trip', () => {
    render(<Trip trip={trip} username={'test'}/>);
    expect(screen.getByText(/Show Trip/)).toBeInTheDocument;
    expect(screen.getByText(/Delete Trip/)).toBeInTheDocument;
});

test('shows details', () => {
    render(<Trip trip={trip}/>);
    fireEvent.click(screen.getByText(/Show Trip/));
    expect(screen.getByText(/testLocName1/)).toBeInTheDocument;
    expect(screen.getByText(/testLocName2/)).toBeInTheDocument;
    expect(screen.queryByText(/Show Trip/)).not.toBeInTheDocument;
    fireEvent.click(screen.getByText(/Hide Trip/));
    expect(screen.queryByText(/testLocName1/)).not.toBeInTheDocument;
    expect(screen.queryByText(/testLocName2/)).not.toBeInTheDocument;
    expect(screen.getByText(/Show Trip/)).toBeInTheDocument;
});