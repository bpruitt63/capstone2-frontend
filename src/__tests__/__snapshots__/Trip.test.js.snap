// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Trip"
  >
    <div
      class="tripTitle"
    >
      <h5>
        <span>
          test1
        </span>
         - 1.00 km
      </h5>
      <p>
        Rating: 1
      </p>
    </div>
    <button
      class="toggle btn btn-secondary"
      type="button"
    >
      Show Trip
    </button>
    <br />
  </div>
</DocumentFragment>
`;
